# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../boost/gcc43.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -aNru boost_1_34_1-ORIGINAL/boost/archive/polymorphic_iarchive.hpp boost_1_34_1/boost/archive/polymorphic_iarchive.hpp
--- boost_1_34_1-ORIGINAL/boost/archive/polymorphic_iarchive.hpp	2005-12-11 00:12:51.000000000 -0600
+++ boost_1_34_1/boost/archive/polymorphic_iarchive.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -17,6 +17,7 @@
 //  See http://www.boost.org for updates, documentation, and revision history.
 
 #include <cstddef> // std::size_t
+#include <climits>
 #include <boost/config.hpp>
 
 #if defined(BOOST_NO_STDC_NAMESPACE)
diff -aNru boost_1_34_1-ORIGINAL/boost/archive/polymorphic_oarchive.hpp boost_1_34_1/boost/archive/polymorphic_oarchive.hpp
--- boost_1_34_1-ORIGINAL/boost/archive/polymorphic_oarchive.hpp	2006-02-11 23:43:06.000000000 -0600
+++ boost_1_34_1/boost/archive/polymorphic_oarchive.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -17,6 +17,7 @@
 //  See http://www.boost.org for updates, documentation, and revision history.
 
 #include <cstddef> // size_t
+#include <climits>
 #include <string>
 
 #include <boost/config.hpp>
diff -aNru boost_1_34_1-ORIGINAL/boost/date_time/tz_db_base.hpp boost_1_34_1/boost/date_time/tz_db_base.hpp
--- boost_1_34_1-ORIGINAL/boost/date_time/tz_db_base.hpp	2005-10-23 15:15:06.000000000 -0500
+++ boost_1_34_1/boost/date_time/tz_db_base.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -158,7 +158,7 @@
       typedef typename time_zone_type::base_type time_zone_base_type;
       typedef typename time_zone_type::time_duration_type time_duration_type;
       typedef time_zone_names_base<char_type> time_zone_names;
-      typedef dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets;
+      typedef boost::date_time::dst_adjustment_offsets<time_duration_type> dst_adjustment_offsets;
       typedef std::basic_string<char_type> string_type;
 
       //! Constructs an empty database
diff -aNru boost_1_34_1-ORIGINAL/boost/mpl/zip_view.hpp boost_1_34_1/boost/mpl/zip_view.hpp
--- boost_1_34_1-ORIGINAL/boost/mpl/zip_view.hpp	2004-09-02 10:40:42.000000000 -0500
+++ boost_1_34_1/boost/mpl/zip_view.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -37,7 +37,7 @@
     typedef zip_iterator<
           typename transform1<
                 IteratorSeq
-              , next<_1>
+              , boost::mpl::next<_1>
             >::type
         > next;
 };
@@ -48,8 +48,8 @@
 struct zip_view
 {
  private:
-    typedef typename transform1< Sequences, begin<_1> >::type first_ones_;
-    typedef typename transform1< Sequences, end<_1> >::type last_ones_;
+    typedef typename transform1< Sequences, boost::mpl::begin<_1> >::type first_ones_;
+    typedef typename transform1< Sequences, boost::mpl::end<_1> >::type last_ones_;
     
  public:
     typedef nested_begin_end_tag tag;
diff -aNru boost_1_34_1-ORIGINAL/boost/regex/v4/perl_matcher.hpp boost_1_34_1/boost/regex/v4/perl_matcher.hpp
--- boost_1_34_1-ORIGINAL/boost/regex/v4/perl_matcher.hpp	2006-10-18 07:55:30.000000000 -0500
+++ boost_1_34_1/boost/regex/v4/perl_matcher.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -18,6 +18,8 @@
 #  include BOOST_ABI_PREFIX
 #endif
 
+#include <climits>
+
 namespace boost{
 namespace re_detail{
 
diff -aNru boost_1_34_1-ORIGINAL/boost/regex/v4/regex_split.hpp boost_1_34_1/boost/regex/v4/regex_split.hpp
--- boost_1_34_1-ORIGINAL/boost/regex/v4/regex_split.hpp	2005-01-21 11:22:38.000000000 -0600
+++ boost_1_34_1/boost/regex/v4/regex_split.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -21,6 +21,8 @@
 #ifndef BOOST_REGEX_SPLIT_HPP
 #define BOOST_REGEX_SPLIT_HPP
 
+#include <climits>
+
 namespace boost{
 
 #ifdef BOOST_HAS_ABI_HEADERS
diff -aNru boost_1_34_1-ORIGINAL/boost/regex/v4/states.hpp boost_1_34_1/boost/regex/v4/states.hpp
--- boost_1_34_1-ORIGINAL/boost/regex/v4/states.hpp	2005-09-20 07:01:25.000000000 -0500
+++ boost_1_34_1/boost/regex/v4/states.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -23,6 +23,8 @@
 #  include BOOST_ABI_PREFIX
 #endif
 
+#include <climits>
+
 namespace boost{
 namespace re_detail{
 
diff -aNru boost_1_34_1-ORIGINAL/boost/serialization/collection_traits.hpp boost_1_34_1/boost/serialization/collection_traits.hpp
--- boost_1_34_1-ORIGINAL/boost/serialization/collection_traits.hpp	2005-06-21 00:19:04.000000000 -0500
+++ boost_1_34_1/boost/serialization/collection_traits.hpp	2008-03-03 03:09:49.000000000 -0600
@@ -22,6 +22,7 @@
 // compiles recognize the same set of primitive types, the possibility
 // exists for archives to be non-portable if class information for primitive
 // types is included.  This is addressed by the following macros.
+#include <climits>
 #include <boost/config.hpp>
 #include <boost/mpl/integral_c.hpp>
 #include <boost/mpl/integral_c_tag.hpp>
+#include <climits>
 
 ///////////////////////////////////////////////////////////////////////////////
 namespace phoenix {
diff -aNru boost_1_34_1-ORIGINAL/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp boost_1_34_1/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp
--- boost_1_34_1-ORIGINAL/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp	2006-12-20 10:38:24.000000000 -0600
+++ boost_1_34_1/boost/wave/cpplexer/re2clex/cpp_re2c_lexer.hpp	2008-03-03 03:08:59.000000000 -0600
@@ -16,6 +16,7 @@
 #include <string>
 #include <cstdio>
 #include <cstdarg>
+#include <cstring>
 #if defined(BOOST_SPIRIT_DEBUG)
 #include <iostream>
 #endif // defined(BOOST_SPIRIT_DEBUG)
diff -aNru boost_1_34_1-ORIGINAL/libs/spirit/test/numerics_tests.cpp boost_1_34_1/libs/spirit/test/numerics_tests.cpp
--- boost_1_34_1-ORIGINAL/libs/spirit/test/numerics_tests.cpp	2006-02-28 16:56:32.000000000 -0600
+++ boost_1_34_1/libs/spirit/test/numerics_tests.cpp	2008-03-03 03:08:59.000000000 -0600
@@ -11,6 +11,7 @@
 #include <boost/spirit/actor/assign_actor.hpp>
 #include <iostream>
 #include <boost/detail/lightweight_test.hpp>
+#include <climits>
 
 using namespace std;
 using namespace boost::spirit;
